src/app/components/book-list/book-list.component.ts

```diff
@@ -80,17 +99,29 @@ export class BookListComponent implements OnInit {
             data: book
         })
 
-        dialogRef.afterClosed().subscribe(result => {
-            if(result) { 
-                this.deleteBook(book);; 
-            }  
-        })
+        this.subscriptions.push(
+            dialogRef.afterClosed().subscribe(result => {
+                if(result) { 
+                    this.deleteBook(book);; 
+                }  
+            })            
+        )
     }
 
     deleteBook(book: IReadBook): void{
-        this.booksService
-            .deleteReadBook(book.id);
+        this.subscriptions.push(
+            this.booksService
+                .deleteReadBook(book.id)
+                .subscribe()       
+        )
     }
 
-    ngOnInit(){this.getBooks()}
+    ngOnInit() {
+        this.subscriptions = [];
+        this.getBooks();
+    }
+
+    ngOnDestroy() {
+        this.subscriptions.forEach(s => s.unsubscribe())
```

Comment:
Pls move this logic and subscription field to abstract class. In other case we would be forced to do the same thing every time.